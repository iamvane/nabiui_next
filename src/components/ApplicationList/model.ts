import { AvailabilityType } from '../Availability/model';

export interface Application {
  applicationId: number;
  instructorId: number;
  applicationMessage: string;
  applicationRate: string;
  availability: AvailabilityType;
  avatar: string;
  displayName: string;
  reviews: number;
  yearsOfExperience: number;
  age: number;
  backgroundCheckStatus: string;

}
export interface ApplicationListType {
  id: number;
  requestTitle: string;
  dateCreated: string;
  applications: Application[];
  freeTrial: boolean;
}
