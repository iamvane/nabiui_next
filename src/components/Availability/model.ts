export interface Availability {
  day: string;
  from: string;
  to: string;
}

export interface AvailabilityType {
  mon8to10?: boolean;
  mon10to12?: boolean;
  mon12to3?: boolean;
  mon3to6?: boolean;
  mon6to9?: boolean;
  tue8to10?: boolean;
  tue10to12?: boolean;
  tue12to3?: boolean;
  tue3to6?: boolean;
  tue6to9?: boolean;
  wed8to10?: boolean;
  wed10to12?: boolean;
  wed12to3?: boolean;
  wed3to6?: boolean;
  wed6to9?: boolean;
  thu8to10?: boolean;
  thu10to12?: boolean;
  thu12to3?: boolean;
  thu3to6?: boolean;
  thu6to9?: boolean;
  fri8to10?: boolean;
  fri10to12?: boolean;
  fri12to3?: boolean;
  fri3to6?: boolean;
  fri6to9?: boolean;
  sat8to10?: boolean;
  sat10to12?: boolean;
  sat12to3?: boolean;
  sat3to6?: boolean;
  sat6to9?: boolean;
  sun8to10?: boolean;
  sun10to12?: boolean;
  sun12to3?: boolean;
  sun3to6?: boolean;
  sun6to9?: boolean;
  [key: string]: any;
}