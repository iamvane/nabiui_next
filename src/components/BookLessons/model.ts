export interface BookLessonPackages {
  name: string;
	lessonNumber: number;
	value: string;
}

export interface BookLessonsPayload {
	studentId: number;
	package: string;
	paymentMethodId?: number;
	paymentMethodCode?: string;
}

export interface BookLessonsData {
	clientSecret?: string;
	lessonRate?: number
	lessonsPrice?: number;
	paymentMethods?: any[]
	placementFee?: number;
	processingFee?: number;
	subTotal?: number
	total?: number;
	freeTrial?: boolean;
	virtuosoDiscount?: number;
	discounts?: number;
	instructor: {
		avatar: string;
		reviews: {
			rate: string;
			quantity: number
		},
		backgroundCheckStatus: string;
		display_name: string;
		rate: number;
		yearsOfExperience: number;
		age: number;
	}
}

export enum LessonStatusType {
	Scheduled = 'scheduled',
	Missed = 'missed',
	Conplete = 'complete'
}

export interface LessonType {
	studentId?: number;
	date?: string;
	time?: string;
	status?: LessonStatusType;
	grade?: number;
	comment?: string;
}
