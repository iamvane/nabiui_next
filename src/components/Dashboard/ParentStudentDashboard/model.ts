export interface StudentDetailsType {
  id?: number;
  name?: string;
  age?: number;
  instrument?: string;
  skillLevel?: string;
  lessonPlace?: string;
  lessonDuration?: string;
  studentId?: number;
  specialNeeds?: string[];
  notes?: string;
}

export interface ParentProfileType {
  students: StudentDetailsType[];
}
