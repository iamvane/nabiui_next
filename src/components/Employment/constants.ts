export namespace EmploymentComponent {
  export enum Text {
    YourEmployment = 'Your Employment',
    ListYourPastExperience = 'List your past experience, make sure it is relevant',
    AddEmployment = 'Add Employment'
  }
}

export namespace EmploymentFormComponent {
  export enum Text {
    AddButton = 'Add',
    SaveChangesButton = 'Save Changes',
    CancelButton = 'Cancel',
    From = 'From',
    To = 'To',
    Present = 'Present'
  }

  export enum FieldNames {
    Employer = 'employer',
    JobTitle = 'jobTitle',
    JobLocation = 'jobLocation',
    FromMonth = 'fromMonth',
    FromYear = 'fromYear',
    ToMonth = 'toMonth',
    ToYear = 'toYear',
    StillWorkHere = 'stillWorkHere'
  }

  export enum Ids {
    Employer = 'employer',
    JobTitle = 'jobTitle',
    JobLocation = 'jobLocation',
    FromMonth = 'fromMonth',
    FromYear = 'fromYear',
    ToMonth = 'toMonth',
    ToYear = 'toYear',
  }

  export enum Placeholders {
    Employer = 'Company/employer name',
    JobTitle = 'Job title',
    JobLocation = 'Job location (city, state, country)',
    SchoolLocation = 'School location (city, state, country)'
  }

  export enum DisabledPlaceholders {
    SelectMonth = 'Select Month',
    SelectYear = 'Select Year'
  }

  export enum Labels {
    ICurrentlyWorkHere = 'I currently work here',
  }
}

export namespace EmploymentAddedComponent {
  export const timeline = '{fromMonthReplace}, {fromYearReplace} - {toMonthReplace}, {toYearReplace}';
  export const timelinePresent = '{fromMonthReplace}, {fromYearReplace} - Present';

  export const fromMonthPlaceholder = '{fromMonthReplace}';
  export const fromYearPlaceholder = '{fromYearReplace}';

  export const toMonthPlaceholder = '{toMonthReplace}';
  export const toYearPlaceholder = '{toYearReplace}';
}
