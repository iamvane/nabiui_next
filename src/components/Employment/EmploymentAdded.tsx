import * as React from 'react';

import {
  Icon,
  IconButton,
  Grid,
  Typography
} from '@material-ui/core';

import { EmploymentType }  from 'components/Employment/model';
import { EmploymentAddedComponent } from 'components/Employment/constants';

interface Props extends EmploymentType {
  deleteEmployment: (employmentId: any) => void;
  editEmployment: (employmentId: any) => void;
  gridWidth?: boolean | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | undefined;
  notEditable?: boolean;
}

/**
 * EmploymentAdded
 */
const EmploymentAdded: React.StatelessComponent<Props> = props => {
  const fromMonth = props.fromMonth.charAt(0).toUpperCase() + props.fromMonth.slice(1);
  const toYear = props.toYear ? props.toYear : '';
  const toMonth = props.toMonth ? props.toMonth : '';
  const toMonthCapital = toMonth.charAt(0).toUpperCase() + toMonth.slice(1);

  return (
    <Grid
      item={true}
      md={props.gridWidth && props.gridWidth}
      xs={12}
      className={`${props.notEditable ? '' : 'nabi-editable-item'} nabi-margin-top-small`}
    >
      <div className="nabi-action-buttons">
        <IconButton
          color="primary"
          className="nabi-float-right nabi-margin-left-xsmall"
          aria-label="Delete"
          onClick={() => props.deleteEmployment(props.id)}
        >
          <Icon>delete</Icon>
        </IconButton>

        <IconButton
          color="primary"
          className="nabi-float-right"
          aria-label="Edit"
          onClick={() => props.editEmployment(props.id)}
        >
          <Icon>edit</Icon>
        </IconButton>
      </div>
      <Typography className="nabi-margin-top-small nabi-text-uppercase" variant="body2">
        <mark>{props.jobTitle}</mark>
      </Typography>
      <Typography>
        <mark>{props.employer}</mark>
      </Typography>
      {props.stillWorkHere || (!props.toYear && !toMonthCapital) ? (
        <Typography>
          <mark>
            {EmploymentAddedComponent.timelinePresent.replace(
              EmploymentAddedComponent.fromMonthPlaceholder,
              fromMonth
            ).replace(
              EmploymentAddedComponent.fromYearPlaceholder,
              props.fromYear,
            )}
          </mark>
        </Typography>
      ) : (
        <Typography>
          <mark>
            {EmploymentAddedComponent.timeline.replace(
              EmploymentAddedComponent.fromMonthPlaceholder,
              fromMonth
            ).replace(
              EmploymentAddedComponent.fromYearPlaceholder,
              props.fromYear,
            ).replace(
              EmploymentAddedComponent.toMonthPlaceholder,
              toMonthCapital,
            ).replace(
              EmploymentAddedComponent.toYearPlaceholder,
              toYear,
            )}
          </mark>
        </Typography>
      )}
      <Typography>
        <mark>{props.jobLocation}</mark>
      </Typography>
    </Grid>
  );
};

export default EmploymentAdded;
