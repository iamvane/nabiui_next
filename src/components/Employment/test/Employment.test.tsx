import * as React from 'react';
import { Employment } from 'components/Employment/Employment';
import { shallow } from 'enzyme';
import { EmploymentType } from 'components/Employment/model';

describe('Employment', () => {
  let wrapper: any;
  const mockFunction = jest.fn();

  const mockEmployment: EmploymentType = {
    id: 0,
    employer: 'Oracle',
    jobTitle: 'Software Engineer',
    jobLocation: 'Cambridge',
    fromMonth: 'september',
    fromYear: '2017',
    toMonth: 'march',
    toYear: '2019'
  };

  beforeEach(() => {
    wrapper = shallow(
      <Employment
        employment={[mockEmployment]}
        fetchEmployment={mockFunction}
        addEmployment={mockFunction}
        editEmployment={mockFunction}
        deleteEmployment={mockFunction}
        isFetchEmploymentRequesting={false}
        isAddEmploymentRequesting={false}
        isDeleteEmploymentRequesting={false}
        isEditEmploymentRequesting={false}
        fetchEmploymentError=""
        addEmploymentError=""
        editEmploymentError=""
        deleteEmploymentError=""
      />
    );
  });

  describe('When the employment form is hidden', () => {
    it('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('When the employment form is displayed', () => {
    beforeEach(() => {
      wrapper.setState({showEmploymentForm: true});
    });

    it('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
