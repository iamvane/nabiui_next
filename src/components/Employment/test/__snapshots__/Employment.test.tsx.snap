// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employment When the employment form is displayed Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Employment
    addEmployment={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
      }
    }
    addEmploymentError=""
    deleteEmployment={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
      }
    }
    deleteEmploymentError=""
    editEmployment={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
      }
    }
    editEmploymentError=""
    employment={
      Array [
        Object {
          "employer": "Oracle",
          "fromMonth": "september",
          "fromYear": "2017",
          "id": 0,
          "jobLocation": "Cambridge",
          "jobTitle": "Software Engineer",
          "toMonth": "march",
          "toYear": "2019",
        },
      ]
    }
    fetchEmployment={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
      }
    }
    fetchEmploymentError=""
    isAddEmploymentRequesting={false}
    isDeleteEmploymentRequesting={false}
    isEditEmploymentRequesting={false}
    isFetchEmploymentRequesting={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <WithStyles(Typography)
            className="nabi-margin-top-xsmall nabi-text-uppercase"
            variant="body2"
          >
            Add Employment
          </WithStyles(Typography)>
          <EmploymentForm
            allFieldsFilled={false}
            employer=""
            fromMonth=""
            fromYear=""
            handleCancel={[Function]}
            handleChange={[Function]}
            handleOnBlur={[Function]}
            handleSave={[Function]}
            isEditing={false}
            jobLocation=""
            jobTitle=""
            stillWorkHere={undefined}
            toMonth=""
            toYear=""
          />
        </div>,
        <Unknown
          backPath="education"
          buttonText="Continue"
          divider={true}
          handleNext={[Function]}
          icon={
            <WithStyles(Icon)>
              arrow_forward
            </WithStyles(Icon)>
          }
          nextPath="screening"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              className="nabi-margin-top-xsmall nabi-text-uppercase"
              variant="body2"
            >
              Add Employment
            </WithStyles(Typography)>,
            <EmploymentForm
              allFieldsFilled={false}
              employer=""
              fromMonth=""
              fromYear=""
              handleCancel={[Function]}
              handleChange={[Function]}
              handleOnBlur={[Function]}
              handleSave={[Function]}
              isEditing={false}
              jobLocation=""
              jobTitle=""
              stillWorkHere={undefined}
              toMonth=""
              toYear=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Add Employment",
              "className": "nabi-margin-top-xsmall nabi-text-uppercase",
              "variant": "body2",
            },
            "ref": null,
            "rendered": "Add Employment",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "allFieldsFilled": false,
              "employer": "",
              "fromMonth": "",
              "fromYear": "",
              "handleCancel": [Function],
              "handleChange": [Function],
              "handleOnBlur": [Function],
              "handleSave": [Function],
              "isEditing": false,
              "jobLocation": "",
              "jobTitle": "",
              "stillWorkHere": undefined,
              "toMonth": "",
              "toYear": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "backPath": "education",
          "buttonText": "Continue",
          "divider": true,
          "handleNext": [Function],
          "icon": <WithStyles(Icon)>
            arrow_forward
          </WithStyles(Icon)>,
          "nextPath": "screening",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <WithStyles(Typography)
              className="nabi-margin-top-xsmall nabi-text-uppercase"
              variant="body2"
            >
              Add Employment
            </WithStyles(Typography)>
            <EmploymentForm
              allFieldsFilled={false}
              employer=""
              fromMonth=""
              fromYear=""
              handleCancel={[Function]}
              handleChange={[Function]}
              handleOnBlur={[Function]}
              handleSave={[Function]}
              isEditing={false}
              jobLocation=""
              jobTitle=""
              stillWorkHere={undefined}
              toMonth=""
              toYear=""
            />
          </div>,
          <Unknown
            backPath="education"
            buttonText="Continue"
            divider={true}
            handleNext={[Function]}
            icon={
              <WithStyles(Icon)>
                arrow_forward
              </WithStyles(Icon)>
            }
            nextPath="screening"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                className="nabi-margin-top-xsmall nabi-text-uppercase"
                variant="body2"
              >
                Add Employment
              </WithStyles(Typography)>,
              <EmploymentForm
                allFieldsFilled={false}
                employer=""
                fromMonth=""
                fromYear=""
                handleCancel={[Function]}
                handleChange={[Function]}
                handleOnBlur={[Function]}
                handleSave={[Function]}
                isEditing={false}
                jobLocation=""
                jobTitle=""
                stillWorkHere={undefined}
                toMonth=""
                toYear=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Add Employment",
                "className": "nabi-margin-top-xsmall nabi-text-uppercase",
                "variant": "body2",
              },
              "ref": null,
              "rendered": "Add Employment",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "allFieldsFilled": false,
                "employer": "",
                "fromMonth": "",
                "fromYear": "",
                "handleCancel": [Function],
                "handleChange": [Function],
                "handleOnBlur": [Function],
                "handleSave": [Function],
                "isEditing": false,
                "jobLocation": "",
                "jobTitle": "",
                "stillWorkHere": undefined,
                "toMonth": "",
                "toYear": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backPath": "education",
            "buttonText": "Continue",
            "divider": true,
            "handleNext": [Function],
            "icon": <WithStyles(Icon)>
              arrow_forward
            </WithStyles(Icon)>,
            "nextPath": "screening",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Employment When the employment form is hidden Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Employment
    addEmployment={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    addEmploymentError=""
    deleteEmployment={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    deleteEmploymentError=""
    editEmployment={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    editEmploymentError=""
    employment={
      Array [
        Object {
          "employer": "Oracle",
          "fromMonth": "september",
          "fromYear": "2017",
          "id": 0,
          "jobLocation": "Cambridge",
          "jobTitle": "Software Engineer",
          "toMonth": "march",
          "toYear": "2019",
        },
      ]
    }
    fetchEmployment={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    fetchEmploymentError=""
    isAddEmploymentRequesting={false}
    isDeleteEmploymentRequesting={false}
    isEditEmploymentRequesting={false}
    isFetchEmploymentRequesting={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <SectionTitle
            text="Your Employment"
          />
          <WithStyles(Typography)
            className="nabi-margin-top-xsmall"
          >
            List your past experience, make sure it is relevant
          </WithStyles(Typography)>
          <ul>
            <li
              className="nabi-list"
            >
              <EmploymentAdded
                deleteEmployment={[Function]}
                editEmployment={[Function]}
                employer="Oracle"
                fromMonth="september"
                fromYear="2017"
                gridWidth={6}
                id={0}
                jobLocation="Cambridge"
                jobTitle="Software Engineer"
                stillWorkHere={undefined}
                toMonth="march"
                toYear="2019"
              />
            </li>
          </ul>
          <div
            className="nabi-margin-top-medium"
          >
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
              variant="contained"
            >
              <WithStyles(Icon)
                className="nabi-margin-right-xsmall"
              >
                add
              </WithStyles(Icon)>
              Add Employment
            </WithStyles(Button)>
          </div>
        </div>,
        <Unknown
          backPath="education"
          buttonText="Continue"
          divider={true}
          handleNext={[Function]}
          icon={
            <WithStyles(Icon)>
              arrow_forward
            </WithStyles(Icon)>
          }
          nextPath="screening"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <SectionTitle
              text="Your Employment"
            />,
            <WithStyles(Typography)
              className="nabi-margin-top-xsmall"
            >
              List your past experience, make sure it is relevant
            </WithStyles(Typography)>,
            <ul>
              <li
                className="nabi-list"
              >
                <EmploymentAdded
                  deleteEmployment={[Function]}
                  editEmployment={[Function]}
                  employer="Oracle"
                  fromMonth="september"
                  fromYear="2017"
                  gridWidth={6}
                  id={0}
                  jobLocation="Cambridge"
                  jobTitle="Software Engineer"
                  stillWorkHere={undefined}
                  toMonth="march"
                  toYear="2019"
                />
              </li>
            </ul>,
            <div
              className="nabi-margin-top-medium"
            >
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
                variant="contained"
              >
                <WithStyles(Icon)
                  className="nabi-margin-right-xsmall"
                >
                  add
                </WithStyles(Icon)>
                Add Employment
              </WithStyles(Button)>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "text": "Your Employment",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "List your past experience, make sure it is relevant",
              "className": "nabi-margin-top-xsmall",
            },
            "ref": null,
            "rendered": "List your past experience, make sure it is relevant",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="nabi-list"
                >
                  <EmploymentAdded
                    deleteEmployment={[Function]}
                    editEmployment={[Function]}
                    employer="Oracle"
                    fromMonth="september"
                    fromYear="2017"
                    gridWidth={6}
                    id={0}
                    jobLocation="Cambridge"
                    jobTitle="Software Engineer"
                    stillWorkHere={undefined}
                    toMonth="march"
                    toYear="2019"
                  />
                </li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": <EmploymentAdded
                    deleteEmployment={[Function]}
                    editEmployment={[Function]}
                    employer="Oracle"
                    fromMonth="september"
                    fromYear="2017"
                    gridWidth={6}
                    id={0}
                    jobLocation="Cambridge"
                    jobTitle="Software Engineer"
                    stillWorkHere={undefined}
                    toMonth="march"
                    toYear="2019"
                  />,
                  "className": "nabi-list",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "deleteEmployment": [Function],
                    "editEmployment": [Function],
                    "employer": "Oracle",
                    "fromMonth": "september",
                    "fromYear": "2017",
                    "gridWidth": 6,
                    "id": 0,
                    "jobLocation": "Cambridge",
                    "jobTitle": "Software Engineer",
                    "stillWorkHere": undefined,
                    "toMonth": "march",
                    "toYear": "2019",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <WithStyles(Button)
                color="primary"
                onClick={[Function]}
                variant="contained"
              >
                <WithStyles(Icon)
                  className="nabi-margin-right-xsmall"
                >
                  add
                </WithStyles(Icon)>
                Add Employment
              </WithStyles(Button)>,
              "className": "nabi-margin-top-medium",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(Icon)
                    className="nabi-margin-right-xsmall"
                  >
                    add
                  </WithStyles(Icon)>,
                  "Add Employment",
                ],
                "color": "primary",
                "onClick": [Function],
                "variant": "contained",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "add",
                    "className": "nabi-margin-right-xsmall",
                  },
                  "ref": null,
                  "rendered": "add",
                  "type": [Function],
                },
                "Add Employment",
              ],
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "backPath": "education",
          "buttonText": "Continue",
          "divider": true,
          "handleNext": [Function],
          "icon": <WithStyles(Icon)>
            arrow_forward
          </WithStyles(Icon)>,
          "nextPath": "screening",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <SectionTitle
              text="Your Employment"
            />
            <WithStyles(Typography)
              className="nabi-margin-top-xsmall"
            >
              List your past experience, make sure it is relevant
            </WithStyles(Typography)>
            <ul>
              <li
                className="nabi-list"
              >
                <EmploymentAdded
                  deleteEmployment={[Function]}
                  editEmployment={[Function]}
                  employer="Oracle"
                  fromMonth="september"
                  fromYear="2017"
                  gridWidth={6}
                  id={0}
                  jobLocation="Cambridge"
                  jobTitle="Software Engineer"
                  stillWorkHere={undefined}
                  toMonth="march"
                  toYear="2019"
                />
              </li>
            </ul>
            <div
              className="nabi-margin-top-medium"
            >
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
                variant="contained"
              >
                <WithStyles(Icon)
                  className="nabi-margin-right-xsmall"
                >
                  add
                </WithStyles(Icon)>
                Add Employment
              </WithStyles(Button)>
            </div>
          </div>,
          <Unknown
            backPath="education"
            buttonText="Continue"
            divider={true}
            handleNext={[Function]}
            icon={
              <WithStyles(Icon)>
                arrow_forward
              </WithStyles(Icon)>
            }
            nextPath="screening"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <SectionTitle
                text="Your Employment"
              />,
              <WithStyles(Typography)
                className="nabi-margin-top-xsmall"
              >
                List your past experience, make sure it is relevant
              </WithStyles(Typography)>,
              <ul>
                <li
                  className="nabi-list"
                >
                  <EmploymentAdded
                    deleteEmployment={[Function]}
                    editEmployment={[Function]}
                    employer="Oracle"
                    fromMonth="september"
                    fromYear="2017"
                    gridWidth={6}
                    id={0}
                    jobLocation="Cambridge"
                    jobTitle="Software Engineer"
                    stillWorkHere={undefined}
                    toMonth="march"
                    toYear="2019"
                  />
                </li>
              </ul>,
              <div
                className="nabi-margin-top-medium"
              >
                <WithStyles(Button)
                  color="primary"
                  onClick={[Function]}
                  variant="contained"
                >
                  <WithStyles(Icon)
                    className="nabi-margin-right-xsmall"
                  >
                    add
                  </WithStyles(Icon)>
                  Add Employment
                </WithStyles(Button)>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "text": "Your Employment",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "List your past experience, make sure it is relevant",
                "className": "nabi-margin-top-xsmall",
              },
              "ref": null,
              "rendered": "List your past experience, make sure it is relevant",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className="nabi-list"
                  >
                    <EmploymentAdded
                      deleteEmployment={[Function]}
                      editEmployment={[Function]}
                      employer="Oracle"
                      fromMonth="september"
                      fromYear="2017"
                      gridWidth={6}
                      id={0}
                      jobLocation="Cambridge"
                      jobTitle="Software Engineer"
                      stillWorkHere={undefined}
                      toMonth="march"
                      toYear="2019"
                    />
                  </li>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": <EmploymentAdded
                      deleteEmployment={[Function]}
                      editEmployment={[Function]}
                      employer="Oracle"
                      fromMonth="september"
                      fromYear="2017"
                      gridWidth={6}
                      id={0}
                      jobLocation="Cambridge"
                      jobTitle="Software Engineer"
                      stillWorkHere={undefined}
                      toMonth="march"
                      toYear="2019"
                    />,
                    "className": "nabi-list",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "deleteEmployment": [Function],
                      "editEmployment": [Function],
                      "employer": "Oracle",
                      "fromMonth": "september",
                      "fromYear": "2017",
                      "gridWidth": 6,
                      "id": 0,
                      "jobLocation": "Cambridge",
                      "jobTitle": "Software Engineer",
                      "stillWorkHere": undefined,
                      "toMonth": "march",
                      "toYear": "2019",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <WithStyles(Button)
                  color="primary"
                  onClick={[Function]}
                  variant="contained"
                >
                  <WithStyles(Icon)
                    className="nabi-margin-right-xsmall"
                  >
                    add
                  </WithStyles(Icon)>
                  Add Employment
                </WithStyles(Button)>,
                "className": "nabi-margin-top-medium",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <WithStyles(Icon)
                      className="nabi-margin-right-xsmall"
                    >
                      add
                    </WithStyles(Icon)>,
                    "Add Employment",
                  ],
                  "color": "primary",
                  "onClick": [Function],
                  "variant": "contained",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "add",
                      "className": "nabi-margin-right-xsmall",
                    },
                    "ref": null,
                    "rendered": "add",
                    "type": [Function],
                  },
                  "Add Employment",
                ],
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backPath": "education",
            "buttonText": "Continue",
            "divider": true,
            "handleNext": [Function],
            "icon": <WithStyles(Icon)>
              arrow_forward
            </WithStyles(Icon)>,
            "nextPath": "screening",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
